common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -340,6 +342,7 @@ object DotcomRenderingDataModel {
       availableTopics: Option[Seq[Topic]],
       newsletter: Option[NewsletterData],
       topicResult: Option[TopicResult],
+      mostPopular: Seq[OnwardCollectionResponse] = Seq(),
```

Comment:
We tend to make optional field optional, such that we don't send them through in the response at all when they don't exists (instead of sending an empty list for example) - just to reduce the size of the payload and to avoid displaying empty things by mistake. 

I also think in this case Option[OnwardCollectionResponse] is more appropriate as you would never have more than one collection in the most read field (a collection contains multiple trails). I think? I think.

Reply:
Ah, I was wondering why we have optional lists. Nice reason. So would this need a check in frontend to see if the list is empty and return None if so? It does feel quite complicated for fields we know will always return a list? (I've amended it anyway to be consistent)

> as you would never have more than one collection in the most read field

This will be a list as it will contain all the data needed for this component, which can consist of multiple data sources. Using the screenshot in the example, it would be:
```
[mostViewedData, mostViewedInSectionData]
```

Reply:
> So would this need a check in frontend to see if the list is empty and return None if so?

I don't properly understand how json serialisation is done in scala/play, but I had assumed that [this check](https://github.com/guardian/frontend/blob/e06ce983e2119552ae7c73f2c0e492bb781352ef/common/app/model/dotcomrendering/DotcomRenderingDataModel.scala#L179-L182) was responsible for filtering out null values (though perhaps it's doing something else).

Reply:
@bryophyta not sure what that link is referring to?

This was done to send less data down the pipes to DCR. As mentioned above, I find the compromise of having to deal with more complex type on the server and the client a bit fiddly, but I am assuming this was taken into account.

When modelling, it is very rare to have an `Option[Seq[T]]`, as that can be more simply represented as `Seq[T]`, and it can just be left empty. `Option` can often be seen as a single item list. The reason for not making an `Seq` optional is it adds extra complexity when trying to access its values - you'll need to `map` twice to get to the values. There's not many cases where that's what you're trying to express.

e.g.

```scala
val maybeList = Some(Seq(1))
maybeList.map(seq => seq.map(num => num + 1))
```

But, if we are going to use the standard JSON codecs that we get with play, an empty `Seq` parses to `[]`, where an empty `Option[Seq[T]]` parsers to not including the key (not `null`).

e.g.
```scala
case class Thing(list: Seq[String], maybeList: Option[Seq[String])

val thing1 = Thing(list = Seq(), maybeList = None)
val thing2 = Thing(list = Seq(), maybeList = Some(Seq()))

println(thing1.toJson())
println(thing2.toJson())
```

```JSONC
// thing1
{
  "list": []
}

// thing2
{
  "list": [],
  "maybeList": []
}
```

Another way to have done this would to not make it the model aware of this choice, but rather add it to the parsers for types that are being sent to DCR. This would remove this complexity from the model, and leave it to the parsing later, leaving a little complexity on the client i.e. having to deal with a missing parameter and an empty list everytime you want to access it e.g.

```jsx
const list = CAPI.onwards;
{list && list.length > 0 &&
  <div class="container">
    <ul>{list.map(item => <li>{item}</li>)}</ul>
  </div>
}
```

Reply:
I would be in favour of having better parsing these things in DCR, and use the appropriate default.

You can even have things like non-empty lists with types:

```ts
type Item = unknown;
type NonEmptyList = [Item, ...Item[]];
```